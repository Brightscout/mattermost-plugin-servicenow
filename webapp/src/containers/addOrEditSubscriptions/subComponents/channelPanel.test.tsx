import React from 'react';
import {Provider} from 'react-redux';

import {shallow, ShallowWrapper} from 'enzyme';

import {configureStore} from '../../../../tests/setup';

import ChannelPanel from './channelPanel';

const mockOnContinue = jest.fn();
const mockSetApiError = jest.fn();
const mockSetApiResponseValid = jest.fn();
const mockSetShowModalLoader = jest.fn();
const mockSetChannel = jest.fn();
const mockSetChannelOptions = jest.fn();

const mockChannelOptions: DropdownOptionType[] = [{
    label: 'Channel 1',
    value: 'Channel 1',
}, {
    label: 'Channel 2',
    value: 'Channel 2',
}, {
    label: 'Channel 3',
    value: 'Channel 3',
}];

const channelPanelProps = {
    className: 'mockClassName',
    error: 'mockError',
    onContinue: mockOnContinue,
    actionBtnDisabled: true,
    setApiError: mockSetApiError,
    setApiResponseValid: mockSetApiResponseValid,
    setShowModalLoader: mockSetShowModalLoader,
    channel: 'mockChannel',
    setChannel: mockSetChannel,
    channelOptions: mockChannelOptions,
    setChannelOptions: mockSetChannelOptions,
};

describe('Channel Panel', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    const initialState = {};
    const mockStore = configureStore();

    beforeEach(() => {
        const store = mockStore(initialState);
        component = shallow(
            <Provider store={store}>
                <ChannelPanel {...channelPanelProps}/>
            </Provider>,
        );
    });

    afterEach(() => {
        component.unmount();
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    // it('Should apply the passed className prop', () => {
    //     expect(component.find(ChannelPanel).hasClass(channelPanelProps.className)).toBeTruthy();
    // });

    // it('Should render the events panel body correctly', () => {
    //     expect(component.find(".Dropdown")).toHaveLength(1);
    // });

    // it('Should render the error correctly', () => {
    //     expect(component.contains(
    //         <ModalSubtitleAndError error={channelPanelProps.error}/>,
    //     )).toBeTruthy();
    // });
});
