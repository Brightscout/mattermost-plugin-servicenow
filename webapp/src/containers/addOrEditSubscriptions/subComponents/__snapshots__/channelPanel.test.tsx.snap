// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel Panel Should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ForwardRef
    actionBtnDisabled={true}
    channel="mockChannel"
    channelOptions={
      Array [
        Object {
          "label": "Channel 1",
          "value": "Channel 1",
        },
        Object {
          "label": "Channel 2",
          "value": "Channel 2",
        },
        Object {
          "label": "Channel 3",
          "value": "Channel 3",
        },
      ]
    }
    className="mockClassName"
    error="mockError"
    onContinue={[MockFunction]}
    setApiError={[MockFunction]}
    setApiResponseValid={[MockFunction]}
    setChannel={[MockFunction]}
    setChannelOptions={[MockFunction]}
    setShowModalLoader={[MockFunction]}
  />
</ContextProvider>
`;
